### PR TITLE
Silence various warnings.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Include missing type signature, un-used imports and un-used fields on records.

### CODE REVIEW

File: conduit-extra/Data/Conduit/Network.hs [hunk 1]
```diff
@@ -38,17 +38,15 @@ module Data.Conduit.Network
...
 import Network.Socket (Socket)
-import Network.Socket.ByteString (sendAll, recv)
+import Network.Socket.ByteString (sendAll)
 import Data.ByteString (ByteString)
 import qualified GHC.Conc as Conc (yield)
 import qualified Data.ByteString as S
-import qualified Data.ByteString.Char8 as S8
 import Control.Monad.IO.Class (MonadIO (liftIO))
-import Control.Exception (throwIO, SomeException, try, finally, bracket, IOException, catch)
+---import Control.Exception (throwIO, SomeException, try, finally, bracket, IOException, catch)
```
Comment (olorin) [inline conduit-extra/Data/Conduit/Network.hs]: Did you mean to leave this commented?

Comment (tmcgilchrist) [inline conduit-extra/Data/Conduit/Network.hs]: Not especially, I'll fix this up.

File: conduit-extra/Data/Conduit/Text.hs [hunk 2]
```diff
@@ -61,7 +60,7 @@ import Data.Streaming.Text
 --
 -- Since 0.3.0
 data Codec = Codec
-    { codecName :: T.Text
+    { _codecName :: T.Text
```
Comment (tmcgilchrist) [inline conduit-extra/Data/Conduit/Text.hs]: I'm not sure why this is being defined in the first place if it's not being used, renaming to an underscore prefix silences the warnings but is quite unsightly. Happy for alternate suggestions on how this should be handled.

File: conduit-extra/Data/Conduit/Text.hs [hunk 3]
```diff
@@ -150,7 +149,7 @@ decodeNew
     -> Int
     -> (B.ByteString -> DecodeResult)
     -> Conduit B.ByteString m T.Text
-decodeNew onFailure name =
+decodeNew onFailure _name =
```
Comment (tmcgilchrist) [inline conduit-extra/Data/Conduit/Text.hs]: Same thing here, it could be `_` or `_name`

File: conduit-extra/Data/Conduit/Text.hs (hunk 2 above)
Comment (snoyberg) [inline conduit-extra/Data/Conduit/Text.hs]: It's a good documentation strategy to indicate what the purpose of the field is.

File: conduit-extra/Data/Conduit/Text.hs (hunk 3 above)
Comment (snoyberg) [inline conduit-extra/Data/Conduit/Text.hs]: Keeping `_name` is similarly nice documentation

**Comment by snoyberg** [comment]
It doesn't look like this solves the initial problem completely:

```
Data/Conduit/Binary.hs:315:28: Not in scope: `<$>'
```

File: conduit-extra/Data/Conduit/Binary.hs [hunk 4]
```diff
@@ -55,7 +55,6 @@ import Control.Monad.Trans.Resource (allocate, release)
 import Control.Monad.Trans.Class (lift)
 import qualified System.IO as IO
 import Data.Word (Word8, Word64)
-import Control.Applicative ((<$>))
```
Comment (snoyberg) [inline conduit-extra/Data/Conduit/Binary.hs]: This is the cause of the build breakage I referenced.

**Comment by tmcgilchrist** [comment]
Apologies, should be ok now.

**Pull request merged**